clang/include/clang/Basic/Attr.td

```diff
@@ -1504,6 +1506,12 @@ def NVPTXKernel : InheritableAttr, TargetSpecificAttr<TargetNVPTX> {
   let Documentation = [Undocumented];
 }
 
+def SPIRKernel : InheritableAttr, TargetSpecificAttr<TargetAnySPIR> {
+  let Spellings = [Clang<"spir_kernel">];
+  let Subjects = SubjectList<[Function]>;
+  let Documentation = [Undocumented];
```

Comment:
Why is this non-documented?  WE very much don't approve/accept new undocumented attributes without good reason.

Reply:
Thanks, I wasn't sure how strict we were with that, will add documentation.